processor/src/services/mock-payment.service.ts

```
@@ -170,9 +189,50 @@ export class MockPaymentService extends AbstractPaymentService {
    * @returns Promise with mocking data containing operation status and PSP reference
    */
   public async refundPayment(request: RefundPaymentRequest): Promise<PaymentProviderModificationResponse> {
+    await this.ctPaymentService.updatePayment({
+      id: request.payment.id,
+      transaction: {
+        type: 'Refund',
+        amount: request.amount,
+        state: 'Success',
+      },
+    });
     return { outcome: PaymentModificationStatus.APPROVED, pspReference: request.payment.interfaceId as string };
   }
 
+  /**
+   * Reverse payment
+   *
+   * @remarks
+   * Abstract method to execute payment reversals in support of automated reversals to be triggered by checkout api. The actual invocation to PSPs should be implemented in subclasses
+   *
+   * @param request
+   * @returns Promise with outcome containing operation status and PSP reference
+   */
+  public async reversePayment(request: ReversePaymentRequest): Promise<PaymentProviderModificationResponse> {
+    const hasCharge = request.payment.transactions.some(
+      (tx: Transaction) => tx.type === 'Charge' && (tx.state === 'Success' || tx.state === 'Pending'),
```

the pending scenario should not be considered. if it is pending we should throw an error. I mean, if there is not any success authorization or success charge, then we should return an invalid operation error.

---

That makes sense

---

Currently i am returning a General Error, with the message `no payment to revert`. Do you prefer I change that to an invalid operation error?